typescript/sdk/src/providers/MultiProvider.ts

```
@@ -143,25 +349,170 @@ export class MultiProvider<
       );
     }
 
-    const intersectionChainMap = pick(this.chainMap, intersection);
+    const intersectionMetadata = pick(this.metadata, intersection);
+    const intersectionProviders = pick(this.providers, intersection);
+    const intersectionSigners = pick(this.signers, intersection);
+    const multiProvider = new MultiProvider(intersectionMetadata);
+    multiProvider.setProviders(intersectionProviders);
+    multiProvider.setSigners(intersectionSigners);
+    if (this.sharedSigner) multiProvider.setSharedSigner(this.sharedSigner);
 
-    const multiProvider = new MultiProvider<IntersectionChain>({
-      ...intersectionChainMap,
-    });
     return { intersection, multiProvider };
   }
 
-  rotateSigner(newSigner: Signer): void {
-    this.forEach((chain, dc) => {
-      this.setChainConnection(chain, {
-        ...dc,
-        signer: newSigner.connect(dc.provider),
-      });
+  /**
+   * Get chain names excluding given chain name
+   */
+  getRemoteChains(name: ChainName): ChainName[] {
+    return this.getChainNames().filter((n) => n !== name);
+  }
+
+  /**
+   * Get an RPC URL for given chain
+   * @throws if chain's metadata or signer has not been set
+   */
+  getRpcUrl(chainNameOrId: ChainName | number): string {
```

This should throw a better error if no url is in the array